lib/yext/api/knowledge_api/account_settings/user.rb

```diff
@@ -0,0 +1,69 @@
+        #           :endpoint: https://api.yext.com/v2/accounts/{accountId}/users
+        #           :path_regex: v2/accounts/[^/]+?/users
+        #           :default_version: 20161012
+        #           :documentation: http://developer.yext.com/docs/api-reference/#operation/getUsers
+        #           :comment: Lists all Users in an account.
+        #           :sandbox_only: false
+        #         - :action: :create
+        #           :method: :post
+        #           :endpoint: https://api.yext.com/v2/accounts/{accountId}/users
+        #           :path_regex: v2/accounts/[^/]+?/users
+        #           :default_version: 20161012
+        #           :documentation: http://developer.yext.com/docs/api-reference/#operation/createUser
+        #           :comment: Create a new User.
+        #           :sandbox_only: false
+        #         - :action: :show
+        #           :method: :get
+        #           :endpoint: https://api.yext.com/v2/accounts/{accountId}/users/{userId}
+        #           :path_regex: v2/accounts/[^/]+?/users/[^/]+?
+        #           :default_version: 20161012
+        #           :documentation: http://developer.yext.com/docs/api-reference/#operation/getUser
+        #           :comment: Retrieves details of a specific User.
+        #           :sandbox_only: false
+        #         - :action: :update
+        #           :method: :put
+        #           :endpoint: https://api.yext.com/v2/accounts/{accountId}/users/{userId}
+        #           :path_regex: v2/accounts/[^/]+?/users/[^/]+?
+        #           :default_version: 20161012
+        #           :documentation: http://developer.yext.com/docs/api-reference/#operation/updateUser
+        #           :comment: Updates an existing User.
+        #           :sandbox_only: false
+        #         - :action: :destroy
+        #           :method: :delete
+        #           :endpoint: https://api.yext.com/v2/accounts/{accountId}/users/{userId}
+        #           :path_regex: v2/accounts/[^/]+?/users/[^/]+?
+        #           :default_version: 20161012
+        #           :documentation: http://developer.yext.com/docs/api-reference/#operation/deleteUser
+        #           :comment: Deletes an existing User.
+        #           :sandbox_only: false
+        #         - :action: :change_password
+        #           :method: :put
+        #           :endpoint: https://api.yext.com/v2/accounts/{accountId}/users/{userId}/password
+        #           :path_regex: v2/accounts/[^/]+?/users/[^/]+?/password
+        #           :default_version: 20161012
+        #           :documentation: http://developer.yext.com/docs/api-reference/#operation/updateUserPassword
+        #           :comment: Updates a User's password.
+        #           :sandbox_only: false
+        class User < Yext::Api::Utils::ApiBase
+          include Yext::Api::Concerns::AccountChild
+
+          def change_password(new_password)
```

Comment:
Its odd to me to have a model with a method that commits data. I would expect that we should have a pattern where user gets model, user updates attributes, user saves. In the save we could have a handler that deals with the odd end points that are not a direct rest model update.

At the very least, methods that put/update should have a bang on them.

Reply:
Again, this was done very early and I didn't revisit.

I'll take a second look at this.